doc/source/exclude_patterns.md

```diff
@@ -0,0 +1,44 @@
+# Exclude Pattern Settings 
+```--exclude``` is an optional command-line argument for [in-toto-run](https://in-toto.readthedocs.io/en/latest/command-line-tools/in-toto-run.html) that uses [PathSpec](http://python-path-specification.readthedocs.io) while ```exclude_pattern``` is an API argument that follows [gitignore-style patterns](https://git-scm.com/docs/gitignore) and may be specified via the ```ARTIFACT_EXCLUDE_PATTERNS``` setting.
+
+The feature prevents given paths and hashes of 'materials' (files before command execution) and 'products' (files after command execution) from being written together with other information (executed command, return value, stdout and stderr) in generating link metadata files. When a given pattern's directory is excluded, all its files and subdirectories are excluded as well.
+
+# Pattern Format
+  - A backslash must be included in front of patterns beginning with a literal hash or exclamation mark. 
+  - The slash is a directory seperator. Separators at beginning or middle (or both) of a pattern are relative to directory level of particular .gitignore file. Separators at the end of the pattern only  match directories. 
+  - Beginning double asterisks match any leading path segments, inner double asterisks match multiple (or zero) inner path segments, and ending double asterisks match any trailing path segments. Single asterisks match everything except a slash. 
+  - Question marks match any one character except a slash.
+
+## Special Patterns
```

Comment:
Curious here, more than sure. Most of the sub-points here have a single bullet point, bar one. Should we use paragraphs there instead?

Reply:
I followed the [official .gitignore documentation ](https://git-scm.com/docs/gitignore/2.20.0) structure of listing the pattern formats. It may be easier to find quickly than if the information would be in paragraph form, but would be happy to fix it up to look more aesthetically pleasing.